rosbag2_tests/test/rosbag2_tests/record_fixture.hpp

```diff
@@ -66,9 +74,11 @@ class RecordFixture : public TemporaryDirectoryFixture
     while (true) {
       try {
         std::this_thread::sleep_for(50ms);  // wait a bit to not query constantly
-        rosbag2_storage_plugins::SqliteWrapper
-          db(database_path_, rosbag2_storage::storage_interfaces::IOFlag::READ_ONLY);
-        return;
+        if (rcpputils::fs::exists(rcpputils::fs::path{database_path_})) {
```

Comment:
This line was the cause of flakiness;
`SqliteWrapper` does not check if a file exists before opening the Db. The result is that the db is implicitly created *if* it does not exist.